SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKDefaultLaunchConfigGenerator.cpp

```diff
@@ -101,9 +102,89 @@ bool WriteLoadbalancingSection(TSharedRef<TJsonWriter<>> Writer, const FName& Wo
+{
+	ULevelEditorPlaySettings* PlayInSettings = GetMutableDefault<ULevelEditorPlaySettings>();
+
+	PlayInSettings->WorkerTypesToLaunch.Empty(InWorkers.Num());
+	for (const auto& Worker : InWorkers)
+	{
+		if (Worker.Value.bAutoNumEditorInstances)
+		{
+			PlayInSettings->WorkerTypesToLaunch.Add(Worker.Key, Worker.Value.WorkerLoadBalancing->GetNumberOfWorkersForPIE());
+		}
+		else
+		{
+			PlayInSettings->WorkerTypesToLaunch.Add(Worker.Key, Worker.Value.NumEditorInstances);
+		}
+	}
+}
+
+bool GetLoadBalancingStrategyFromWorldSettings(UWorld& World, USpatialOSAbstractLoadBalancingStrategy*& OutStrategy, FIntPoint& OutWorldDimension)
+{
+	const ASpatialWorldSettings* WorldSettings = Cast<ASpatialWorldSettings>(World.GetWorldSettings());
+
+	if (!WorldSettings)
+	{
+		UE_LOG(LogSpatialGDKDefaultLaunchConfigGenerator, Error, TEXT("Missing SpatialWorldSettings on map %s"), *World.GetMapName());
+		return false;
+	}
+
+	if (!WorldSettings->LoadBalanceStrategy)
+	{
+		UE_LOG(LogSpatialGDKDefaultLaunchConfigGenerator, Error, TEXT("Missing Load balancing strategy on map %s"), *World.GetMapName());
+		return false;
+	}
+
+	FSpatialGDKEditorModule& EditorModule = FModuleManager::GetModuleChecked<FSpatialGDKEditorModule>("SpatialGDKEditor");
+
+	if (!EditorModule.GetLBStrategyExtensionManager().GetDefaultLaunchConfiguration(WorldSettings->LoadBalanceStrategy->GetDefaultObject<UAbstractLBStrategy>(), OutStrategy, OutWorldDimension))
+	{
+		UE_LOG(LogSpatialGDKDefaultLaunchConfigGenerator, Error, TEXT("Could not get the SpatialOS Load balancing strategy from %s"), *WorldSettings->LoadBalanceStrategy->GetName());
+		return false;
+	}
+
+	return true;
 }
 
-bool GenerateDefaultLaunchConfig(const FString& LaunchConfigPath, const FSpatialLaunchConfigDescription* InLaunchConfigDescription)
+bool FillWorkerConfigurationFromCurrentMap(TMap<FName, FWorkerTypeLaunchSection>& OutWorkers, FIntPoint& OutWorldDimensions)
+{
+	if (GEditor == nullptr || GEditor->GetWorldContexts().Num() == 0)
+	{
+		return false;
+	}
+
+	const USpatialGDKSettings* SpatialGDKSettings = GetDefault<USpatialGDKSettings>();
+	const USpatialGDKEditorSettings* SpatialGDKEditorSettings = GetDefault<USpatialGDKEditorSettings>();
+
+	UWorld* EditorWorld = GEditor->GetEditorWorldContext().World();
+	check(EditorWorld != nullptr);
+
+	USpatialOSSingleWorkerStrategy* DefaultStrategy = USpatialOSSingleWorkerStrategy::StaticClass()->GetDefaultObject<USpatialOSSingleWorkerStrategy>();
+	USpatialOSAbstractLoadBalancingStrategy* LoadBalancingStrat = DefaultStrategy;
+
+	if (SpatialGDKSettings->bEnableUnrealLoadBalancer)
+	{
+		GetLoadBalancingStrategyFromWorldSettings(*EditorWorld, LoadBalancingStrat, OutWorldDimensions);
+	}
+
+	for (auto& WorkerType : SpatialGDKSettings->ServerWorkerTypes)
+	{
+		FWorkerTypeLaunchSection Conf;
+		if (const FWorkerTypeLaunchSection* WorkerConf = SpatialGDKEditorSettings->LaunchConfigDesc.ServerWorkersMap.Find(WorkerType))
+		{
+			Conf = *WorkerConf;
+		}
```

Comment:
Can we just iterate through SpatialGDKEditorSettings->LaunchConfigDesc.ServerWorkersMap instead?

Reply:
Added a check that both maps are in sync in ValidateGeneratedLaunchConfig so I can iterate on the Editor setting's map.